# Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона

__1. Контрольная работа состоит из пяти вопросов. На каждый вопрос приведено четыре варианта ответа, один из которых правильный. Составьте закон распределения числа правильных ответов при простом угадывании. Найдите M(X), D(X).__

Вероятность ответить верно на $k$ вопросов из $n$: $P(X=k)=C^k_n*p^k*(1-p)^{n-k}$, где $p=\dfrac{1}{4}$ (шанс ответить на один вопрос из $4$ вариантов ответа)

Посчитаем $P(X=k)$ для $k=0,1,2,3,4,5$

In [1]:
import numpy as np

def comb(n, k):
    return np.math.factorial(n) / (np.math.factorial(k) * np.math.factorial(n - k))

In [2]:
n = 5
k = range(0, 6)
P = np.zeros((6,))
for elm in k:
    P[elm] = comb(n, elm)*0.25**elm*(1-0.25)**(n-elm)
print(*P)
print(f'Проверка:{P.sum()==1}')

0.2373046875 0.3955078125 0.263671875 0.087890625 0.0146484375 0.0009765625
Проверка:True


<table>
<tr><th>$x$</th><td>0</td><td>1</td><td>2</td><td>3</td><td>4</td><td>5</td></tr>
<tr><th>$P(X=x)$</th><td>$0.2373046875$</td><td>$0.3955078125$</td><td>$0.263671875$</td><td>$0.087890625$</td><td>$0.0146484375$</td><td>$0.0009765625$</td></tr>
</table>

Найдем мат.ожидание: $M(X) = \sum x_ip_i$ и дисперсию $D(X) = M \left( (X - M(X))^2 \right)$

In [3]:
M = np.dot(k, P).sum()
print(f"Мат.ожидание = {M}")
D = np.dot((k - M)**2, P).sum()
print(f"Дисперсия = {D}")

Мат.ожидание = 1.25
Дисперсия = 0.9375


__2. Вероятность попадания в цель при одном выстреле равна 0.01. Какова вероятность того, что число попаданий при 200 выстрелах будет не менее 5 и не более 10?__

По формуле Бернулли находим вероятности для каждого количества попаданий от $5$ до $10$ (т.к. искомое количество попаданий $x \in [5, 10]$) и суммируем их

In [4]:
n = 200
p = 0.01
P = 0
k_st = 5
k_fn = 10

for elm in range(k_st, k_fn+1):
    P += comb(n, elm)*p**elm*(1-p)**(n-elm)
    
print(f"P = {P} = {P:%}")

P = 0.051739381860374606 = 5.173938%


__3. В условиях предыдущей задачи сколько выстрелов нужно сделать чтобы быть уверенным с вероятностью 0.9, что хотя бы 10 раз будет совершено попадание?__

__Подсказка. Как посчитать P(X >= k) для величины X, имеющей распределение Пуассона? Здесь предстоит немножко покодить.__

Необходимо найти количество выстрелов $(n)$, при котором сумма вероятностей всех возможных попаданий $(k_i)$ строго больше $9$ раз, будет больше или равна $0.9$. Формула Пуассона $P(X = k) = \dfrac{a^k e ^{-a}}{k!}$, где $a=np$.

In [5]:
k = 10
n = 10
p = 0.01

while True:
    P = 0
    
    for k_i in range(k, n+1):
        try:
            P_i = ((n*p)**k_i)*np.exp(-n*p)/np.math.factorial(k_i)
        except OverflowError:
            P_i = 0
        P += P_i
        
    if P >= 0.9:
        break
    n += 1
        
print(f'Количество выстрелов : {n}')

Количество выстрелов : 1421
